catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/config/mapping/ConfluenceConfigMappingProvider.java

```diff
@@ -0,0 +1,154 @@
+
+  private static final String INCLUDE_ARCHIVED_SPACES = "includeArchivedSpaces";
+
+  private static final String INCLUDE_PAGE_CONTENT = "includePageContent";
+
+  private static final String SHORTNAME = "shortname";
+
+  private static final String USERNAME = "username";
+
+  private static final String PASSWORD = "password";
+
+  private static final String CONFLUENCE_SOURCE_FACTORY_PID = "Confluence_Federated_Source";
+
+  private static final Long DEFAULT_POLL_INTERVAL = new Long(60000);
+
+  private static final String EXPANDED_SECTIONS = "expandedSections";
+
+  private static final String[] DEFAULT_EXPANDED_SECTIONS = {
+    "metadata.labels",
+    "space",
+    "history.contributors.publishers.users",
+    "history.lastUpdated",
+    "restrictions.read.restrictions.group",
+    "restrictions.read.restrictions.user"
+  };
+
+  @Override
+  public boolean isPartial() {
+    return false;
+  }
+
+  @Override
+  public boolean canProvideFor(ConfigMapping.Id id) {
+    final boolean canProvideFor = id.getName().equals(CONFLUENCE_SOURCE_FACTORY_PID);
+    LOGGER.debug(
+        "Can Config Mapping Provider [{}] provide a configuration for [{}]? {}",
+        ConfluenceConfigMappingProvider.class.getName(),
+        id.getName(),
+        canProvideFor);
+    return canProvideFor;
+  }
+
+  @Override
+  public Map<String, Object> provide(ConfigMapping.Id id, ConfigService configService)
+      throws ConfigMappingException {
+    final String instanceId = getInstanceId(id);
+    final ConfluenceSourceConfig config = getConfluenceSourceConfig(configService, instanceId);
+
+    final URL url = config.getUrl();
+    final String username = config.getUsername();
+    final String password = config.getPassword();
+
+    final Map<String, Object> properties = new HashMap<>();
+    properties.put(ENDPOINT_URL, url.toString());
+    properties.put(ID, instanceId);
+    properties.put(SHORTNAME, instanceId);
+    properties.put(USERNAME, username);
+    properties.put(PASSWORD, password);
+    properties.put(INCLUDE_ARCHIVED_SPACES, Boolean.FALSE);
+    properties.put(INCLUDE_PAGE_CONTENT, Boolean.FALSE);
+    properties.put(EXPANDED_SECTIONS, DEFAULT_EXPANDED_SECTIONS);
+    properties.put(AVAILABILITY_POLL_INTERVAL, DEFAULT_POLL_INTERVAL);
+    properties.put(BODY_EXPANSION, DEFAULT_BODY_EXPANSION);
+    setExcludedSpaces(config, properties);
+    setAttributeOverrides(config, properties);
+
+    LOGGER.debug(
+        "Returning properties map [{}] for Confluence Source [{}].",
+        properties.toString(),
+        instanceId);
+
+    return Collections.unmodifiableMap(properties);
+  }
+
+  private String getInstanceId(ConfigMapping.Id id) {
+    return id.getInstance()
+        .orElseThrow(() -> new ConfigMappingException("No instance id found in Config Mapping."));
+  }
+
+  private ConfluenceSourceConfig getConfluenceSourceConfig(
+      ConfigService configService, String instanceId) {
+    return configService
```

Comment:
❓ I think we should just returned an empty map if we cannot find the corresponding config object.

Reply:
fixed